chart/templates/serviceaccount.yaml

```diff
@@ -9,11 +9,157 @@ metadata:
   {{- toYaml . | nindent 4 }}
   {{- end }}
 
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "keptn-gitops-operator.serviceAccountName" . }}-get-secrets
+  labels:
+    {{- include "keptn-gitops-operator.labels" . | nindent 4 }}
+  {{- with .Values.serviceAccount.annotations }}
+  annotations:
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "keptn-gitops-operator.serviceAccountName" . }}-get-configmaps
+  labels:
+    {{- include "keptn-gitops-operator.labels" . | nindent 4 }}
+  {{- with .Values.serviceAccount.annotations }}
+  annotations:
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  verbs:
+  - get
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```

Comment:
Is there a plan to make this namespaced as well? Doesn't make much sense to me if the namespaced operator is used, which can still access all keptnservices and keptnprojects in the whole cluster.

Reply:
One option to tackle this is to use a variable `namepaced` in the helm chart and uptionally create a `Role`s or `ClusterRole`s depending on the value of the variable.

Reply:
Good Point! I'll take a look at the version from today.